device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -1216,4 +1221,101 @@ private void verifyMethodsAreSupported()
             throw new UnsupportedOperationException(METHODS_OVER_HTTP_ERROR_MESSAGE);
         }
     }
+
+    /**
+     * Sends the TelemetryMessage to IoT hub.
+     * @param telemetryMessage The user supplied telemetry message.
+     */
+    public void sendTelemetry(TelemetryMessage telemetryMessage)
+    {
+        sendTelemetry(telemetryMessage, null, null);
+    }
+
+    /**
+     * Sends the TelemetryMessage to IoT hub.
+     * @param telemetryMessage The user supplied telemetry message.
+     * @param callback the callback to be invoked when a response is received. Can be {@code null}.
+     * @param callbackContext a context to be passed to the callback. Can be {@code null} if no callback is provided.
+     */
+    public void sendTelemetry(TelemetryMessage telemetryMessage, IotHubEventCallback callback, Object callbackContext)
```

Comment:
What do you think about appending "Async" to the end of these methods that execute callbacks to notify you when the operation has completed?